webviz_subsurface/plugins/_simulation_time_series/utils/from_timeseries_cumulatives.py

```diff
@@ -141,3 +146,25 @@ def rename_vector_from_cumulative(vector: str, as_rate: bool) -> str:
         if as_rate
         else f"INTVL_{vector}"
     )
+
+
+# pylint: disable=too-many-return-statements
+def datetime_to_intervalstr(date: datetime.datetime, freq: Frequency) -> Optional[str]:
+    if date is None:
+        return date
+
+    if freq == Frequency.DAILY:
+        return f"{date.year}-{date.month:02d}-{date.day:02d}"
+    if freq == Frequency.WEEKLY:
+        # Note: weekyear may differ from actual year for week numbers 1,52,53.
+        (weekyear, week, _) = date.isocalendar()
+        return f"{weekyear}-W{week:02d}"
+    if freq == Frequency.MONTHLY:
+        return f"{date.year}-{date.month:02d}"
+    if freq == Frequency.QUARTERLY:
+        return f"{date.year}-Q{(date.month//3+1)}"
```

Comment:
```suggestion
        return f"{date.year}-Q{1 + (date.month-1)//3}"
```

Reply:
Good catch! In my use case I had already ensured that we only use the first month of the quarter, so the `-1` wasn't really necessary, but I agree that it is more robust for later use cases :)